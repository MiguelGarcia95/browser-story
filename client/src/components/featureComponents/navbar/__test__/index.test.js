import React from 'react';
import ReactDOM from 'react-dom';
import Navbar from '../index';

import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import renderer from 'react-test-renderer';

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement('div');
  ReactDOM.render(<Navbar></Navbar>, div);
});

it("renders button correctly", () => {
  const {getByTestId} = render(<Navbar name='hi!'></Navbar>);
  expect(getByTestId('navbar')).toHaveTextContent("hi!");
})

it("renders button correctly", () => {
  const {getByTestId} = render(<Navbar name='delete!'></Navbar>);
  expect(getByTestId('navbar')).toHaveTextContent("delete!");
})

it("matches snapshot 1", () => {
  const tree = renderer.create(<Navbar name='delete!'></Navbar>).toJSON();
  expect(tree).toMatchSnapshot();
})

it("matches snapshot 2", () => {
  const tree = renderer.create(<Navbar name='ha2!'></Navbar>).toJSON();
  expect(tree).toMatchSnapshot();
})