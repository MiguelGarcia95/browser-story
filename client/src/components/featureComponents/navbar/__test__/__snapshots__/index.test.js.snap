// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div
  className="sc-bdVaJa clcVJW"
  data-testid="navbar"
>
  delete!
</div>
`;

exports[`matches snapshot 2 1`] = `
<div
  className="sc-bdVaJa clcVJW"
  data-testid="navbar"
>
  ha2!
</div>
`;
